src/coins.cpp

```diff
@@ -1696,9 +1671,11 @@ libzendoomc::ScFieldElement CCoinsViewCache::GetActiveCertDataHash(const uint256
     if (pSidechain == nullptr)
         return libzendoomc::ScFieldElement{};
 
-    int currentEpochSafeguard = pSidechain->GetCertSubmissionWindowEnd(pSidechain->EpochFor(this->GetHeight())) - pSidechain->creationData.withdrawalEpochLength;
+    int currentEpoch = pSidechain->EpochFor(this->GetHeight()+1);
+    int currentEpochSafeguard = pSidechain->GetCertSubmissionWindowEnd(currentEpoch) - pSidechain->creationData.withdrawalEpochLength;
 
-    if (GetHeight() < currentEpochSafeguard)
+    if (this->GetHeight() < currentEpochSafeguard &&
+        currentEpoch == (pSidechain->lastTopQualityCertReferencedEpoch + 1))
```

Comment:
int certReferencedEpoch = pSidechain->EpochFor(this->GetHeight()+1 - pSidechain->GetSubmissionWindowLength()) - 1;
if(pSidechain->lastTopQualityCertReferencedEpoch == certReferencedEpoch) return lastTopQCDH;
else if(pSidechain->lastTopQualityCertReferencedEpoch -1 == certReferencedEpoch) return pastTopQCDH;
else return error;